lib/liquid/block_body.rb

```diff
@@ -70,49 +70,57 @@ def render(context)
       output = []
       context.resource_limits.render_score += @nodelist.length
 
-      @nodelist.each do |token|
-        # Break out if we have any unhanded interrupts.
+      idx = 0
+      while node = @nodelist[idx]
+        render_node_to_output(node, output, context)
         break if context.interrupt?
-
-        begin
-          # If we get an Interrupt that means the block must stop processing. An
-          # Interrupt is any command that stops block execution such as {% break %}
-          # or {% continue %}
-          if token.is_a?(Continue) || token.is_a?(Break)
-            context.push_interrupt(token.interrupt)
-            break
-          end
-
-          node_output = render_node(token, context)
-
-          unless token.is_a?(Block) && token.blank?
-            output << node_output
-          end
-        rescue MemoryError => e
-          raise e
-        rescue UndefinedVariable, UndefinedDropMethod, UndefinedFilter => e
-          context.handle_error(e, token.line_number)
-          output << nil
-        rescue ::StandardError => e
-          line_number = token.is_a?(String) ? nil : token.line_number
-          output << context.handle_error(e, line_number)
-        end
+        idx += 1
       end
 
       output.join
     end
 
     private
 
+    def render_node_to_output(node, output, context)
+      case node
+      when String
+        node_output = node
```

Comment:
Is this extra variable necessary? Why not
```ruby
check_resources(context, node)
output << node
```
?

Reply:
It's noI necessary. I probably left it there because of the symmetry with the other branches. Should I change it?